pkg/controllers/resources/pods/validatePodSecurity.go

```diff
@@ -0,0 +1,156 @@
+package pods
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/loft-sh/vcluster/pkg/controllers/podsecurity"
+	synccontext "github.com/loft-sh/vcluster/pkg/controllers/syncer/context"
+	admissionv1 "k8s.io/api/admission/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/pod-security-admission/admission"
+	admissionapi "k8s.io/pod-security-admission/admission/api"
+	"k8s.io/pod-security-admission/api"
+	"k8s.io/pod-security-admission/metrics"
+	"k8s.io/pod-security-admission/policy"
+)
+
+func (s *podSyncer) isPodSecurityStandardsValid(ctx *synccontext.SyncContext, pod *corev1.Pod) (bool, error) {
+	result, err := s.validatePodSecurityStandards(ctx.Context, pod)
+	fmt.Println(result)
+	if err != nil {
+		ctx.Log.Errorf(err.Error())
+	} else if result != nil {
+		if !result.Allowed {
+			ctx.Log.Errorf("%s pod creation not allowed: %s", pod.Name, result.Result.Message)
+			pod.Status.Message = result.Result.Message
+			pod.Status.Reason = string(result.Result.Reason)
```

Comment:
Don't update the pod status here please as this can lead to unexpected results, emitting an event is enough for our case